File: webapi/src/main/java/dev/hekmyr/olids/api/dto/ReservationDetailDTO.java
package dev.hekmyr.olids.api.dto;

import dev.hekmyr.olids.api.entity.ReservationDetail;
import java.time.LocalDateTime;
import java.util.UUID;

public class ReservationDetailDTO {

  private UUID id;

  private float pricePerNight;

  private LocalDateTime dateStayStart;

  private LocalDateTime dateStayEnd;

  public ReservationDetailDTO() {}

  public ReservationDetailDTO(ReservationDetail reservationDetail) {
    this.id = reservationDetail.getId();
    this.pricePerNight = reservationDetail.getPricePerNight();
    this.dateStayStart = reservationDetail.getDateStayStart();
    this.dateStayEnd = reservationDetail.getDateStayEnd();
  }

  public UUID getId() {
    return id;
  }

  public void setId(UUID id) {
    this.id = id;
  }

  public float getPricePerNight() {
    return pricePerNight;
  }

  public void setPricePerNight(float pricePerNight) {
    this.pricePerNight = pricePerNight;
  }

  public LocalDateTime getDateStayStart() {
    return dateStayStart;
  }

  public void setDateStayStart(LocalDateTime dateStayStart) {
    this.dateStayStart = dateStayStart;
  }

  public LocalDateTime getDateStayEnd() {
    return dateStayEnd;
  }

  public void setDateStayEnd(LocalDateTime dateStayEnd) {
    this.dateStayEnd = dateStayEnd;
  }
}

File: webapp/src/app/dto/reservation-detail.dto.ts
import { ReservationDetail } from '../interface/reservation-detail.interface';

export class ReservationDetailDTO {
  constructor(
    private _id: string,
    private _pricePerNight: number,
    private _dateStayStart: Date,
    private _dateStayEnd: Date,
    private _isCancelled: boolean
  ) {}

  public get id(): string {
    return this._id;
  }

  public set id(value: string) {
    this._id = value;
  }

  public get pricePerNight(): number {
    return this._pricePerNight;
  }

  public set pricePerNight(value: number) {
    this._pricePerNight = value;
  }

  public get dateStayStart(): Date {
    return this._dateStayStart;
  }

  public set dateStayStart(value: Date) {
    this._dateStayStart = value;
  }

  public get dateStayEnd(): Date {
    return this._dateStayEnd;
  }

  public set dateStayEnd(value: Date) {
    this._dateStayEnd = value;
  }

  public get isCancelled(): boolean {
    return this._isCancelled;
  }

  public set isCancelled(value: boolean) {
    this._isCancelled = value;
  }

  public static fromInterface(
    reservationDetail: ReservationDetail
  ): ReservationDetailDTO {
    return new ReservationDetailDTO(
      reservationDetail.id,
      reservationDetail.pricePerNight,
      reservationDetail.dateStayStart,
      reservationDetail.dateStayEnd,
      reservationDetail.isCancelled
    );
  }
}
